### DISCUSSION TITLE
Dealing with returning customers in a subscription business

### DISCUSSION THREAD

**Discussion by Andrew-Cramp** [CamDavidsonPilon/lifelines#1296] [Q&A] [answered]
Hello, I'm looking to fit a customer survival curve to my subscription business, however I have a number of edge cases that I'm seeking some guidance on how to handle.

Customers can subscribe to our service (birth) and unsubscribe (death). The challenge is there are a number of cases where a customer has been a subscriber for a period of time, then unsubscribed and remained inactive for a period of time before re-subscribing again.

E.g. A customer subscribed for 12 months during 2019, then remained inactive for 12 months during 2020, before subscribing again in January 2021. How do you handle this case?

Thanks!

**Comment by CamDavidsonPilon** [comment, marked as answer]
Hi @karl-mp, the google-able term is called "recurrent events". It requires more general models than what's provided in lifelines. A few shortcuts though:
1. Treat them as multiple observations. If you are doings statistical analysis, and care about std. errors, I suggest clustering std. errors. See https://lifelines.readthedocs.io/en/latest/Examples.html#correlations-between-subjects-in-a-cox-model
2. If the "break" is short, like a month or two, just combine them into a single duration. 
3. Reframe the question: maybe you scope to "time to _first_ churn".

**Reply by Andrew-Cramp** [reply]
Hi @CamDavidsonPilon appreciate your response. Thanks for the suggestions, glad to see a couple of them align with my thoughts.
